### DISCUSSION TITLE
TOUCH_CS pin not defined

### DISCUSSION THREAD

**Discussion by CplMulder** [witnessmenow/ESP32-Cheap-Yellow-Display#85] [General]
Hi there,

I have managed to load the "Hello World" sketch to my CYD after replaceing the UserSetup.h file in the TFT_eSPI folder and that works fine

However I tried the Basics>TouchTest example and although it builds, I get a warning during the build : 

TFT_eSPI.h:973:8: warning: #warning >>>>------>> TOUCH_CS pin not defined, TFT_eSPI touch functions will not be available!

Once the sketch is loaded to my CYD, the screen just shows : 

Pressure=4095
X = 0
Y = 0

and also no response from the touch screen. Have I missed a line to set un the UserSetup.H (thought the CYD version would set the board up incl. the touch screen)

Any help appreciated.

**Comment by witnessmenow** [comment]
Hello, we can't use tft_espi for the touch screen so that warning is to be expected.

It should work though, can you just confirm your board number (the name on the back of the board)

**Reply by CplMulder** [reply]
Thanks for the reply ...

The board is a ESP32-2432S028 (it is missing the R) but does have a touch screen as the loaded demo was touch capable before I flashed over it - is this one on Amazon : 

https://www.amazon.co.uk/dp/B0CJV9HV8G?psc=1&ref=ppx_yo2ov_dt_b_product_details

** EDIT the pics are amazon are incorrect... the RGB led is on the reverse like the one on your GitHub wiki

Thanks a million...

**Reply by slayer522** [reply]
this is why I wanted to look at the pre-installed code from the CYD before flashing it. Still looking..

**Comment by iamsako** [comment]
I'm having this exact same issue. Same board number, hello world example works, the touchscreen test example does not. Getting the same warning and the screen shows the same readings. No response from touching the screen.

I just got the board today and was very tired but wanted to try setting it up quickly. I haven't done any troubleshooting yet besides searching the issue and seeing this post. (I don't know where to begin with troubleshooting to be honest, I'm still somewhat new to microcontrollers and display libraries and such.) I'm glad to see it's not just me with this issue.

**Comment by esacape3d** [comment]
Use HSPI instead of VSPI in the example. 

`SPIClass mySpi = SPIClass(HSPI);`

Hope this helps.

**Reply by CplMulder** [reply]
Thanks for your reply @esacape3d 

Do you know what version of TFT_eSPI and what XPT2046 you are using... I am on the latest versions, using the CYD UserSetup.h and TouchTest.ino - and there is no definition forHSPI (or VSPI for that matter): 

`2-TouchTest:56:27: error: 'HSPI' was not declared in this scope
 SPIClass mySpi = SPIClass(HSPI);
`

There is also no begin method for SPIClass accepting one parameter a few lines further down ?

`2-TouchTest:65:66: error: no matching function for call to 'SPIClass::begin(int, int, int, int)'
   mySpi.begin(XPT2046_CLK, XPT2046_MISO, XPT2046_MOSI, XPT2046_CS);`

Feels like I have the incorrect library or summin .... perhaps do you know where this is defined so I can see what I am missing ?

Thanks a million mate!

**Reply by esacape3d** [reply]
My version is as below. 
TFT_eSPI: 2.5.34
XPT2046_Touchscreen: 1.4 

I found the HSPI is defined in esp32-hal-spi.h
```
#if CONFIG_IDF_TARGET_ESP32C3 || CONFIG_IDF_TARGET_ESP32S3
#define FSPI  0
#define HSPI  1
#else
#define FSPI  1 //SPI bus attached to the flash (can use the same data lines but different SS)
#define HSPI  2 //SPI bus normally mapped to pins 12 - 15, but can be matrixed to any pins
#if CONFIG_IDF_TARGET_ESP32
#define VSPI  3 //SPI bus normally attached to pins 5, 18, 19 and 23, but can be matrixed to any pins
#endif
#endif
```

But interestingly, I just realized the original example with VSPI is compiled without an error for now. I might miss some other things.. FYI, the below is the last few lines of the compilation. 

```
Using library SPI at version 2.0.0 in folder: 
Using library XPT2046_Touchscreen at version 1.4 in folder: 
Using library TFT_eSPI at version 2.5.34 in folder: 
Using library FS at version 2.0.0 in folder: 
Using library SPIFFS at version 2.0.0
```

**Reply by CplMulder** [reply]
@esacape3d you are a legend !! - versions were the same and changing to HSPI worked fine.... turns out the undefined symbols was because I had the incorrect board selected...all working now!! 

Thanks again.

@iamsako try this fix!!

**Reply by iamsako** [reply]
Sorry for the late reply, I only got back around to tinkering with the display again yesterday. @esacape3d Your fix worked, thank you for the help!

**Comment by gpena208777** [comment]
Same problem here
